### DISCUSSION TITLE
What version number to expect when building from Master source or Latest Release?

### DISCUSSION THREAD

**Discussion by fade2gray** [dnGrep/dnGrep#687] [Q&A] [answered]
The contents of the Master and Latest Release sources (3.0.64.0) appear identical when compared using WinMerge, but when querying 'About' from dnGrep's menu after building both, I'm informed the version number is 3.0.0.0?

**Comment by doug24** [comment, marked as answer]
Pull requests, merge builds, and release builds are all done on AppVeyor, which increments the build number - the third digit of the version.  The version number in the code is just a placeholder that is replaced by the build script - see the first line in appveyor.yml in the solution directory.  The latest release is the 64th build since I changed the version to 3.0 and reset the build number, and it's now up to 66.

The only time the version number matters in a local dev build is when testing the installer kit and you need to set a higher build number than the last release build you have installed.  When doing that I usually change the local version number and then discard the change.

**Reply by fade2gray** [reply]
So, is it just a matter of changing the version number in `appveyor.yml`, but making it one version higher than seen in [AppVeyor/History](https://ci.appveyor.com/project/JVimes/dngrep/history), i.e. 3.0.69.0?

**Reply by doug24** [reply]
Setting the version number is fully automatic for all build done on AppVeyor.  Don't do anything, don't set the value.

Only if you are testing changes to the installer kit on your local box before committing the installer changes, then you need to make the version number at least one higher than your current installed version - otherwise the test installation will not be installed.  Once done with testing, do not commit the build number change.

**Reply by fade2gray** [reply]
Ok, I think I'll leave AppVeyor alone for now, and just tinker with with my local build and make any suggested changes via the usual channels.
